### PR TITLE
Add Microsoft.Maui.Graphics to the MAUI workload

### PR DESCRIPTION
**Pull request by pjcollins**
`Microsoft.Maui.Graphics` and
`Microsoft.Maui.Graphics.Win2D.WinUI.Desktop` have been added to the
`maui-core` and `maui-windows` workloads as library pack dependencies.
These changes should improve the usability of MAUI with Visual Studio
Internal Previews by reducing the number of NuGet feeds required.

The `sign.yml` template has been updated to download the Maui Graphics
package version specified in `eng/Versions.props` from the `dotnet6`
feed before running the .msi generation tooling in dotnet/arcade.  The
following additional .msi files will now be produced the build, and can
be inserted into Visual Studio:

 * Microsoft.Maui.Graphics.$(PackageVersion)-x64.msi
 * Microsoft.Maui.Graphics.$(PackageVersion)-x86.msi
 * Microsoft.Maui.Graphics.Win2D.$(PackageVersion)-x64.msi
 * Microsoft.Maui.Graphics.Win2D.$(PackageVersion)-x86.msi

This will allow `Microsoft.Maui.Graphics` and
`Microsoft.Maui.Graphics.Win2D.WinUI.Desktop` to be installed from a
local `library-pack` folder when restoring a .NET MAUI project:

     Restoring packages for C:\Users\Peter\source\repos\MauiApp6\MauiApp6\MauiApp6.csproj...
     Installed Microsoft.Maui.Graphics 6.0.200-preview.14.1092+sha.027fa21-azdo.5841062 from C:\Program Files\dotnet\library-pack

### CODE REVIEW

**Comment by pjcollins** [comment]
I set up a VS PR build to test the initial changes, and a fresh restore looks like it is grabbing Microsoft.Maui.Graphics from the local library-pack path:

         Restoring packages for C:\Users\Peter\source\repos\MauiApp6\MauiApp6\MauiApp6.csproj...
         Installed Microsoft.Maui.Dependencies 6.0.300-rc.1.5290+sha.c6483283d-azdo.5935992 from C:\Program Files\dotnet\library-packs
         Installed Microsoft.Maui.Extensions 6.0.300-rc.1.5290+sha.c6483283d-azdo.5935992 from C:\Program Files\dotnet\library-packs
         Installed Microsoft.Maui.Graphics 6.0.200-preview.14.1092+sha.027fa21-azdo.5841062 from C:\Program Files\dotnet\library-pack

I'm going to try to test this with a version that is not already in NuGet.org to be more confident, but it seems like this approach should work for us.

**Comment by pjcollins** [comment]
Test build that produced the Graphics .msi files: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5945372&view=results

File: eng/pipelines/common/sign.yml [hunk 1]
```diff
@@ -22,4 +24,28 @@ stages:
...
+                -Source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json
+          - task: NuGetCommand@2
+            displayName: Download Microsoft.Maui.Graphics.Win2D.WinUI.Desktop
+            inputs:
+              command: custom
+              arguments: >-
+                install Microsoft.Maui.Graphics.Win2D.WinUI.Desktop -Prerelease -Version $(MauiGraphicsVersion)
+                -OutputDirectory $(Build.StagingDirectory)\maui-graphics
+                -Source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json
+                -Source https://api.nuget.org/v3/index.json
```
Comment (jonathanpeppers) [inline eng/pipelines/common/sign.yml:45]: Do we need both sources here?

File: src/Workload/Microsoft.NET.Sdk.Maui/Microsoft.NET.Sdk.Maui.csproj [hunk 2]
```diff
@@ -63,12 +63,19 @@
...
+        OldValue="@MAUI_GRAPHICS_VERSION@"
+        NewValue="$(MicrosoftMauiGraphicsVersion)"
+    />
     <ReplaceText
         Input="%(_JsonVariableMatrix.OutputPath)"
         Output="%(_JsonVariableMatrix.OutputPath)"
         OldValue="%(_JsonVariableMatrix.OldValue)"
         NewValue="%(_JsonVariableMatrix.NewValue)"
     />
+    <Message Text="##vso[task.setvariable variable=MicrosoftMauiGraphicsVersion;isOutput=true]$(MicrosoftMauiGraphicsVersion)" Importance="high" />
```
Comment (jonathanpeppers) [inline src/Workload/Microsoft.NET.Sdk.Maui/Microsoft.NET.Sdk.Maui.csproj:78]: I don't want to hold this up, b/c we probably need this in ASAP.

It looks like it was tricky to pass this value from MSBuild to .yaml. Could we have made Maui's build output the Maui.Graphics `.nupkg` files to something like `artifacts\signed`? and as long as they got copied to `nugets\signed`, that might have been a little simpler?

**Review by jonathanpeppers — APPROVED** [review]
This looks good, I just wondered about the extra `-Source`.

File: src/Workload/Microsoft.NET.Sdk.Maui/Microsoft.NET.Sdk.Maui.csproj (hunk 2 above)
Comment (rmarinho) [reply src/Workload/Microsoft.NET.Sdk.Maui/Microsoft.NET.Sdk.Maui.csproj:78]: why can we read it from Version.props ?

**Comment by rmarinho** [comment]
Failing tests are normal for now..

**Review by Redth — APPROVED** [review]

File: eng/pipelines/common/sign.yml (hunk 1 above)
Comment (pjcollins) [reply eng/pipelines/common/sign.yml:45]: The NuGet.org source was needed to install some of the Microsoft.Maui.Graphics.Win2D.WinUI.Desktop dependencies that were not on the dotnet6 feed.

**Pull request merged**